### ISSUE TITLE
compatibility with chart.js 3.0

### ISSUE THREAD

**Issue by rashearth** [AbelHeinsbroek/chartjs-plugin-crosshair#58]
Hi, Thanks for this wonderfull app, its really great to use.
I have one request.
Since the chart.js 3.0 is ongoing in good condition, it would be wonderful if you could make it compatible on 
Chart.js 3.0 as well.

Could you please work it out on it?

Thank you & Regards,
Rashearth

**Comment by rashearth** [comment]
Hi, the chart.js 3.0 beta have launched version9!
If possible , could you kindly work out the compatibility?

Just a gentle reminder!

**Comment by rashearth** [comment]
Hi, this is just a gentle reminder, is this script now compatible with chart.js  > 3.0? 

Best Regards,
Rashearth

**Comment by Yohandah** [comment]
This project looks dead to me sadly

**Comment by marcusnewton** [comment]
Man this plugin would be really nice to have right about now

**Comment by AbelHeinsbroek** [comment]
@marcusnewton I've added a chartjs3 branch with some preliminary fixes to support ChartJS 3.4. I have some issues with syncing zoom events that I still need to fix though. Can you check if that version is working for you?

**Comment by marcusnewton** [comment]
@AbelHeinsbroek I tried it but you don't seem to be using the latest version of Chartjs https://github.com/AbelHeinsbroek/chartjs-plugin-crosshair/blob/chartjs3/package.json#L47

**Comment by AbelHeinsbroek** [comment]
Whoops. Mind trying again?

**Comment by marcusnewton** [comment]
@AbelHeinsbroek I'm honestly just getting `Chart.register is not a function` here https://github.com/AbelHeinsbroek/chartjs-plugin-crosshair/blob/chartjs3/src/trace.js#L550

**Comment by marcusnewton** [comment]
Perhaps it should be imported as an ES module

**Comment by AbelHeinsbroek** [comment]
hi @marcusnewton would you mind trying again?

**Comment by marcusnewton** [comment]
@AbelHeinsbroek can't test right now but i'll get back to you :)

**Comment by vividos** [comment]
@AbelHeinsbroek @marcusnewton I just added PR #78 that adds a GitHub Action to build the plugin. This way the plugin can be tested without having npm or the build tools installed. I tested the build output, and the plugin works, together with Chart.js 3.4.0. I'm only using the crosshair, not zooming or panning, though.

**Comment by marcusnewton** [comment]
Yeah I dunno why this doesn't work for me. Everything builds fine, I import the plugin into my Svelte component but don't see `crosshair` in the plugins registry. Again, I would probably suggest updating the usage to work as an ES module import, letting the user register the plugin themselves. This might be some compatibility issues when using this with certain module bundlers like rollup, but I'm not sure.

**Comment by vividos** [comment]
@AbelHeinsbroek I checked the chartjs3 branch and it seems the change from PR #49 still needs to be applied there (it's just a typo when checking `xScaleType`.

**Comment by roma2341** [comment]
> > Yeah I dunno why this doesn't work for me. Everything builds fine, I import the plugin into my Svelte component but don't see `crosshair` in the plugins registry.
> 
> I’m having the same issue with this plugin v1.2.0 in an Angular implementation with rollup.
> 
> 'crosshair' does not exist in type '_DeepPartialObject<PluginOptionsByType>'

Have you solved the issue ?

**Comment by DaveSkender** [comment]
Nope.  I turned it off.  It's either internally not registered correctly with newer Chart.js versions, or there's some TypeScript incompatibility or missing typings issue.  I've not tried to find a fix to contribute here either.  The Owner is slow to respond or not actively managing this project, from what I can see.
